src/wp-includes/class-wp-scripts.php

```diff
@@ -871,8 +883,22 @@ private function get_dependents( $handle ) {
 	 */
 	private function get_intended_strategy( $handle ) {
 		$script_args = $this->get_data( $handle, 'script_args' );
+		$strategy    = isset( $script_args['strategy'] ) ? $script_args['strategy'] : false;
 
-		return isset( $script_args['strategy'] ) ? $script_args['strategy'] : false;
+		if ( $strategy && ! $this->is_valid_strategy( $strategy ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				sprintf(
+					/* translators: 1: $strategy, 2: $handle */
+					__( 'Invalid strategy `%1$s` set for `%2$s`, during script registration.' ),
+					$strategy,
+					$handle
+				),
+				'6.3.0'
```

Comment:
Are we 100% certain this will be in 6.3.0? If so, we can also update all our `@since` docblock params accordingly?

Reply:
@ 10upsimon I am only 70 to 80 % sure, but this was something I needed to add to get this function right. So added it. I'll note this on our Asana task for `@since`.